CBMC: Add proof and spec for `polyvecl_pointwise_acc_montgomery` and `polyvec_matrix_pointwise_montgomery`

mkannwischer opened the pull request:
- Resolves https://github.com/pq-code-package/mldsa-native/issues/121
- Resolves https://github.com/pq-code-package/mldsa-native/issues/138

mkannwischer commented:
The `polyvecl_pointwise_acc_montgomery` proof is now officially our slowest proof. 
It takes 196s, 389s, 557s in CI for the 3 parameter sets. 

Unless someone already has an idea how to improve that, I think we can address that in a follow-up.

jakemas commented:
Nice one! I was attempting this today (https://github.com/pq-code-package/mldsa-native/tree/polyvecl_pointwise_acc_montgomery), nice to see the result!

Funny that CI is failing because NISTs FIPS 204 host is down..!

jakemas commented:
my only recommendation would be the docustring:

https://github.com/pq-code-package/mldsa-native/pull/267

mkannwischer commented:
> my only recommendation would be the docustring: (i'm trying to fill each missing one out as I go, it also supports the proof nicely)
> 
> https://github.com/pq-code-package/mldsa-native/pull/270/files#diff-727bef316d30bd39ded691eb13c5bd41cfeb5761b4bdb91c6c03fe76c062f2a4R621-R633
> 
> ```
> /*************************************************
>  * Name:        polyvec_matrix_pointwise_montgomery
>  *
>  * Description: Compute matrix-vector multiplication in NTT domain with
>  *              pointwise multiplication and multiplication by 2^{-32}.
>  *              Input matrix and vector must be in NTT domain representation
>  *              with coefficients bounded by MLD_NTT_BOUND. Output vector t
>  *              will have coefficients bounded by MLDSA_Q.
>  *
>  * Arguments:   - polyveck *t: pointer to output vector t
>  *              - const polyvecl mat[MLDSA_K]: pointer to input matrix
>  *              - const polyvecl *v: pointer to input vector v
>  **************************************************/
> ```

Good point. Added. Note that only one input needs to be bound by NTT_BOUND. I corrected that.
